v1.13.0: Update CHANGELOG.md (#2856)
bottlerocket-os/bottlerocket: CHANGELOG.md

# v1.13.0 (2022-03-??)

## OS Changes

* Set NVMe IO request timeouts for EBS according to AWS recommendations ([#2820])
* Add `ethtool` to release ([#2829])

> * Add `NoOpMigration` for migrations and migration-helpers

Reviewer: This can probably be dropped since we added it to v1.12.0 before the release happened. So as far as consumers of Bottlerocket releases go, this was added in the last release and no need to report it again in this release.

  for when no data migration is necessary _(Also cherry-picked to v1.12.0)_ ([#2749])
* Add info logging to migrator in order to document onging migrations ([#2751])

> * Rename `ncurses` to `libncurses` ([#2769])

Reviewer: Is this something a consumer of Bottlerocket will care about, or just an internal detail? I think it's just internal.

* Add `CAP_SYS_MODULE` and `CAP_CHROOT` to bootstrap containers ([#2772])

> * Upgrade `tokio` to latest LTS in Bottlerocket sources and "latest" in tools ([#2736])

Reviewer: This seems like an internal detail too. I guess I'm unclear - is the target of this CHANGELOG users of Bottlerocket or developers of Bottlerocket. If just users, then I think there are a few things in here that may just be confusing and unnecessary to someone just trying to run Bottlerocket.

* Add support for cgroup v2 to NVIDIA variants ([#2802])
* Disable IA and SafeSetID LSM for kernel-5.15 ([#2789])
* Update kernel-5.10 to 5.10.165 and kernel-5.15 to 5.15.90 ([#2795])
* Allow `=` in bootconfig values ([#2806])

> * Upgrade `go-restful` code dependency used by containerd ([#2813])

Reviewer: This one might be worth stating why a containerd dependency is being updated separately from containerd, to address the security issue that is not a security issue.

* Upgraded Rust code dependencies ([#2816])
* Upgraded Go code dependencies ([#2828]) ([#2826])
* Third party package upgrades ([#2825])

## Orchestrator Changes

### Kubernetes

* Change systemd `kublet` service to have restart policy `always` ([#2774])
* Update `eni-max-pod` mappings to include the latest instance types
* Remove Kubernetes 1.21 variants ([#2700])
* Update to Kubernetes v1.25.6 ([#2782])
* Update to Kubernetes v1.24.10 ([#2790])
* Update to Kubernetes v1.23.16 ([#2791])

> * Update to latest EKS-D Kubernetes 1.22 including necessary patches ([#2792])

Reviewer: This should probably state the 1.22 bugfix version like the previous few lines.

# v1.13.0 (2022-03-15)

## OS Changes

> * Set NVMe IO request timeouts for EBS according to AWS recommendations ([#2820])

Reviewer 2: This should be in an AWS platform section.

* Add `ethtool` to Bottlerocket ([#2829])

> * Create and set-up data partition at runtime ([#2807], [#2879], [#2873])

Reviewer 2: ```suggestion
* Support an alternate data partition at runtime ([#2807], [#2879], [#2873])
```

But possibly this should go in the AWS platform section instead, as:
```
* Support an alternate data partition on EC2 instances launched with a single volume
```

> * Create and set-up data partition at runtime ([#2807], [#2879], [#2873])

Reviewer 3: Sorry, on second thought "create" isn't accurate since we do create the partition in `rpm2img`.
```suggestion
* Set-up data partition at runtime ([#2807], [#2879], [#2873])
```

> * Add info logging to migrator in order to document onging migrations ([#2751])

Reviewer 2: ```suggestion
* Improve logging in `migrator` to track ongoing migrations ([#2751])
```

> * Add info logging to migrator in order to document onging migrations ([#2751])

Reviewer 4: `s/onging/ongoing/`

* Add `CAP_SYS_MODULE` and `CAP_CHROOT` to bootstrap containers ([#2772])
* Add support for cgroup v2 ([#2875], [#2802])
* Disable IA and SafeSetID LSM for kernel-5.15 ([#2789])
* Update kernel-5.10 to 5.10.165 and kernel-5.15 to 5.15.90 ([#2795])
* Allow `=` in bootconfig values ([#2806])
* Include `systemd-analyze plot` for `logdog` ([#2880])
* Update host containers ([#2864])

> * Upgrade `go-restful` code dependency used by containerd to avert unaffected CVE ([#2813])

Reviewer 2: nit: this might be too minor a change to mention - I would just lump it in with the catch-all dependency updates for Go, Rust.

* Update third party packages ([#2825], [#2842])

## Orchestrator Changes

### Kubernetes

* Add Kubernetes 1.26 variants ([#2771], ([#2876])

> * Change systemd `kubelet` service to have restart policy `always` ([#2774])

Reviewer 2: ```suggestion
* Change `kubelet` service to have restart policy `always` ([#2774])
```

> * Update `eni-max-pod` mappings to include the latest AWS instance types ([#2818])

Reviewer 2: nit: could put this in the AWS platform section

> * Remove Kubernetes 1.21 variants ([#2700])

Reviewer 2: maybe lead the Kubernetes section with this item, and bold it

* Update to Kubernetes v1.25.6 ([#2782])
* Update to Kubernetes v1.24.10 ([#2790])
* Update to Kubernetes v1.23.16 ([#2791])
* Update Kubernetes 1.22.17 to include latest EKS-D patches ([#2792])

## Platform Changes

### VMware

> * Removes `k8s.gcr.io` in favor of `public.ecr.aws` ([#2861], ([#2786])

Reviewer 2: Nit: verb tense should align for all these changelog entries:
``suggestion
* Remove `k8s.gcr.io` in favor of `public.ecr.aws` ([#2861], ([#2786])
```

* Disable UDP offload for primary interface ([#2850])

### ECS

> * Enable FireLens capability in `aws-ecs-1` variant ([#2819])

Reviewer 2: ECS is an orchestrator, not a platform.

## Build Changes

* Ensure empty build/rpms directory is included in build context ([#2784])

> * `buildsys`, `systemd`: add `unified-cgroup-hierarchy` image feature ([#2845])

Reviewer 2: ```suggestion
* Add image feature flag for cgroup v2 ([#2845])
```

* Enable `systemd-networkd` development via build flag ([#2741], [#2832], [#2750])
* Fix `clippy` linter warnings in source files and add `clippy` CI coverage ([#2745])
* Use `clippy` provided in SDK image ([#2793]) ([#2868])
* Remove unnecessary `time` 0.1.x dependency ([#2748], [#2851])
* Remove unnecessary patch from `containerd` ([#2755])

> * Update `Release.toml` to v1.13.0 ([#2797])

Reviewer 2: nit: can exclude this:
```suggestion
```

* Update Bottlerocket SDK to v0.30.2 ([#2866], [#2857], [#2836])
* Remove outdated `rust_2018_idioms` enforcement ([#2837])

> * Update rust edition to `2021` ([#2835])

Reviewer 2: nit: consistent capitalization for Rust and Go:
```suggestion
* Update Rust edition to `2021` ([#2835])
```

* Upgraded Rust code dependencies ([#2816], [#2869], [#2851], [#2736])
* Upgraded Go code dependencies ([#2828], [#2826])
* Rename `ncurses` to `libncurses` ([#2769])
* Update schnauzer's registry map ([#2867])

## Testing Changes

* Add support for Kubernetes workloads in `testsys` ([#2830])
* Add support for a `tests` directory ([#2737], [#2775])
* Provide advanced config controls to `testsys` ([#2799])
* Fix incorrect migration starting image for VMware testing in `testsys` ([#2804])
* Use testsys v0.0.6 ([#2865])

## Documentation Changes

* Add boot sequence documentation ([#2735])
* Update Bottlerocket version in provisioning step in `PROVISIONING-METAL.md` ([#2785])

> * Add userdata example for setting container registry credentials in `README.md` ([#2803])

Reviewer 2: ```suggestion
* Add user-data example for setting container registry credentials in `README.md` ([#2803])
```

# v1.13.0 (2022-03-15)

> ## OS Changes

Reviewer 5: We should include #2863 here as well.

Maybe "Improve random-access read performance of root volume on some devices"?

Author: Whoops - did have that but thought it was part of Erikson's work with rpm2img as well

* Add `ethtool` to Bottlerocket ([#2829])
* Improve logging in `migrator` to track ongoing migrations ([#2751])
* Improve random-access read performance of root volume on some devices ([#2863])
* Add `CAP_SYS_MODULE` and `CAP_CHROOT` to bootstrap containers ([#2772])
* Add support for cgroup v2 ([#2875], [#2802])
* Disable IA and SafeSetID LSM for kernel-5.15 ([#2789])
* Update kernel-5.10 to 5.10.165 and kernel-5.15 to 5.15.90 ([#2795])
* Allow `=` in bootconfig values ([#2806])
* Include `systemd-analyze plot` for `logdog` ([#2880])
* Update host containers ([#2864])
* Update third party packages ([#2825], [#2842])

## Orchestrator Changes

### Kubernetes

> 

Reviewer 2: nit: extra newline
```suggestion
```

# v1.13.0 (2022-03-15)

## OS Changes

* Set NVMe IO request timeouts for EBS according to AWS recommendations ([#2820])

> * Add `ethtool` to release ([#2829])

Reviewer 6: For clarity, I might reword this line to "Add `ethtool` to packages", or even more general "Add `ethtool` to Bottlerocket"

* Add info logging to migrator in order to document onging migrations ([#2751])
* Add `CAP_SYS_MODULE` and `CAP_CHROOT` to bootstrap containers ([#2772])
* Add support for cgroup v2 ([#2875]) ([#2802])
* Disable IA and SafeSetID LSM for kernel-5.15 ([#2789])
* Update kernel-5.10 to 5.10.165 and kernel-5.15 to 5.15.90 ([#2795])
* Allow `=` in bootconfig values ([#2806])
* Ensure 'prepare-boot' runs after 'mark-successful-boot' ([#2873])
* Include `systemd-analyze plot` for `logdog` ([#2880])
* Update host containers ([#2864])
* Upgrade `go-restful` code dependency used by containerd to avert unaffected CVE ([#2813])
* Third party package upgrades: glibc, hotdog, tcpdump, systemd,
  readline, policycoreutils, open-vm-tools, nvidia-k8s-device-plugin, libpcre
  nvidia-container-toolkit, zstd, libxcrypt, libsepol, libsemanage, libselinux
  libnvidia-container, liblzma, libglib, libdbus, libcap, libaudit, kexec-tools
  iputils, iptables, ecs-agent, ecr-credential-provider, e2fsprogs, dbus-broker
  containerd, ca-certificates, aws-signing-helper, aws-iam-authenticator
  ([#2825])

## Orchestrator Changes

### Kubernetes

* Add Kubernetes 1.26 variants ([#2771]) ([#2876])

> * Change systemd `kublet` service to have restart policy `always` ([#2774])

Reviewer 3: ```suggestion
* Change systemd `kubelet` service to have restart policy `always` ([#2774])
```

* Add Kubernetes 1.26 variants (pending)
* Remove Kubernetes 1.21 variants ([#2700])
* Update to Kubernetes v1.25.6 ([#2782])
* Update to Kubernetes v1.24.10 ([#2790])
* Update to Kubernetes v1.23.16 ([#2791])
* Update Kubernetes 1.22.17 to include latest EKS-D patches ([#2792])
* Update `cni` and `cni-plugin` packages ([#2842])

## Platform Changes

### VMware

* Removes `k8s.gcr.io` in favor of `public.ecr.aws` ([#2861]) ([#2786])
* Disable UDP offload for primary interface ([#2850])

### ECS

* Enable FireLens capability in `aws-ecs-1` variant ([#2819])

## Build Changes

* Ensure empty build/rpms directory is included in build context ([#2784])
* `buildsys`, `systemd`: add `unified-cgroup-hierarchy` image feature ([#2845])
* Enable `systemd-networkd` development via build flag ([#2741]) ([#2832]) ([#2750])
* Fix `clippy` linter warnings in source files and add `clippy` CI coverage ([#2745])
* Use `clippy` provided in SDK image ([#2793]) ([#2868])
* Remove unnecessary `time` 0.1.x dependency ([#2748])
* Remove unnecessary patch from `containerd` ([#2755])
* Update `Release.toml` to v1.13.0 ([#2797])

> * Update Bottlerocket SDK to v0.30.1 ([#2866]) ([#2857]) ([#2836])

Reviewer 7: ```suggestion
* Update Bottlerocket SDK to v0.30.2 ([#2866]) ([#2857]) ([#2836])
```

# v1.13.0 (2022-03-15)

## OS Changes

* Set NVMe IO request timeouts for EBS according to AWS recommendations ([#2820])
* Add `ethtool` to Bottlerocket ([#2829])

> * rpm2img: stripe root filesystem for measureable performance improvement during boot ([#2863]) ([#2879])

Reviewer 3: [#2879](https://github.com/bottlerocket-os/bottlerocket/pull/2879) is related to "Label `BOTTLEROCKET-DATA` at runtime"

> * Label `BOTTLEROCKET-DATA` at runtime ([#2807])

Reviewer 3: ```suggestion
* Create and set-up data partition at runtime ([#2807], [#2879], [#2873])
```

Author: Nice - thanks for the clarity on this one: I wasn't sure the best one-line explanation for it.

* Add info logging to migrator in order to document onging migrations ([#2751])
* Add `CAP_SYS_MODULE` and `CAP_CHROOT` to bootstrap containers ([#2772])

> * Add support for cgroup v2 ([#2875]) ([#2802])

Reviewer 3: nit: 
```suggestion
* Add support for cgroup v2 ([#2875], [#2802])
```

* Disable IA and SafeSetID LSM for kernel-5.15 ([#2789])
* Update kernel-5.10 to 5.10.165 and kernel-5.15 to 5.15.90 ([#2795])
* Allow `=` in bootconfig values ([#2806])

> * Ensure 'prepare-boot' runs after 'mark-successful-boot' ([#2873])

Reviewer 3: This PR should be listed in the same line item as "Label `BOTTLEROCKET-DATA` at runtime ([#2807])"

* Include `systemd-analyze plot` for `logdog` ([#2880])
* Update host containers ([#2864])
* Upgrade `go-restful` code dependency used by containerd to avert unaffected CVE ([#2813])
* Third party package upgrades: glibc, hotdog, tcpdump, systemd,
  readline, policycoreutils, open-vm-tools, nvidia-k8s-device-plugin, libpcre
  nvidia-container-toolkit, zstd, libxcrypt, libsepol, libsemanage, libselinux
  libnvidia-container, liblzma, libglib, libdbus, libcap, libaudit, kexec-tools
  iputils, iptables, ecs-agent, ecr-credential-provider, e2fsprogs, dbus-broker
  containerd, ca-certificates, aws-signing-helper, aws-iam-authenticator

>   ([#2825])

Reviewer 3: Probably keep this simple and people who'd like to see more details can check the PR.
```suggestion
* Update third-party software packages ([#2825])
```

Author: Sure!

## Orchestrator Changes

### Kubernetes

* Add Kubernetes 1.26 variants ([#2771]) ([#2876])
* Change systemd `kublet` service to have restart policy `always` ([#2774])
* Update `eni-max-pod` mappings to include the latest AWS instance types ([#2818])
* Remove Kubernetes 1.21 variants ([#2700])
* Update to Kubernetes v1.25.6 ([#2782])
* Update to Kubernetes v1.24.10 ([#2790])
* Update to Kubernetes v1.23.16 ([#2791])
* Update Kubernetes 1.22.17 to include latest EKS-D patches ([#2792])

> * Update `cni` and `cni-plugin` packages ([#2842])

Reviewer 3: This falls under the third-party software package update line item.
